BENCHMARKING.md

```diff
@@ -45,45 +136,160 @@ A combination of the above approaches is also possible (e.g. it may be useful to
+    (e.g. V100, RadeonVII, ...).
+* `SEGMENTS=<N>` - Split the benchmarked matrix space into `<N>` segments. If
+    specified, `SEGMENT_ID` also has to be set.
+* `SEGMENT_ID=<I>` - used in combination with the `SEGMENTS` variable. `<I>`
+    should be an integer between 1 and `<N>`, the number of `SEGMENTS`. If
+    specified, only the `<I>`-th segment of the benchmark suite will be run.
+* `MATRIX_LIST_FILE=/path/to/matrix_list.file` - allows to list SuiteSparse
+    matrix id or name to benchmark. As an example, a matrix list file containing
+    the following will ensure that benchmarks are ran for only those three
+    matrices:
+    ```
+    1903
+    Freescale/circuit5M
+    thermal2
+    ```
+
+### 4: Publishing the results on Github and analyze the results with the GPE (optional)
+
+The previous experiments generated json files for each matrices, each containing
+timing, iteration count, achieved precision, ... depending on the type of
+benchmark run. These files are available in the directory
+`${ginkgo_build_dir}/benchmark/results/`. These files can be analyzed and
+processed through any tool (e.g. python). In this section, we describe how to
+generate the plots by using Ginkgo's
+[GPE](https://ginkgo-project.github.io/gpe/) tool. First, we need to publish the
+experiments into a Github repository which will be then linked as source input
+to the GPE. For this, we can simply fork the ginkgo-data repository. To do so,
+we can go to the github repository and use the forking interface:
+https://github.com/ginkgo-project/ginkgo-data/
+
+Once it's done, we want to clone the repository locally, put all
+results online and access the GPE for plotting the results. Here are the
+detailed steps:
+```bash
+git clone https://github.com/<username>/ginkgo-data.git $HOME/ginkgo_benchmark/ginkgo-data
+# send the benchmarked data to the ginkgo-data repository
+rsync -rtv ${ginkgo_build}/benchmark/results/ $HOME/ginkgo_benchmark/ginkgo-data/data/
+cd ${HOME}/ginkgo_benchmark/ginkgo-data/data/
+# The following updates the main `.json` files with the list of data.
+# Ensure a python 3 installation is available.
+./build-list . > list.json
+./agregate < list.json > agregate.json
+git config --local user.name "<Name>"
+git config --local user.email "<email>"
+git commit -am "Ginkgo benchmark ${BENCHMARK} of ${SYSTEM_NAME}..."
+git push
+```
+
+Note that depending on what data is of interest, you may need to update the
+scripts `build-list` or `agregate` to change which files you want to agglomerate
+and summarize (depending on the system name), or which data you want to select
+(solver results, spmv results, ...).
+
+For the generating the plots in the GPE, here are the steps to go through:
+1. Access the GPE: https://ginkgo-project.github.io/gpe/
+2. Update data root URL, from
+   https://raw.githubusercontent.com/ginkgo-project/ginkgo-data/master/data to
+   https://raw.githubusercontent.com/<username>/ginkgo-data/<branch>/data
```

Comment:
use code block?
The current one will only show https://raw.githubusercontent.com/<username>/ginkgo-data/<branch>/data